File: R/R/IA_fit.R
IA_fit  =  function( abc, y, g ){
    if (is.null( dim( abc ) )) {
        abc  =  array( abc, dim = c( 3, length( abc ) / 3 ) )
    }
    if (is.null( dim( y ) )) {
        y  =  array( y, dim = c( length( y ), 1 ) )
    }
    if (is.null( dim( g ) )) {
        g  =  array( g, dim = c( length( g ), 1 ) )
    }

    a  =  abc[ 1, , drop = FALSE ]
    b  =  abc[ 2, , drop = FALSE ]
    c  =  abc[ 3, , drop = FALSE ]

    s2  =  1 - a^2 - b^2 - c^2
    d1  =  array( 1, dim = c( length( y ), 1 ) )
    d2  =  array( 1, dim = c( 1, length( a ) ) )

    v0  =  0.5 * colSums( ( y %*% d2 - g %*% a )^2 / ( ( d1 %*% b + g %*% c )^2 + d1 %*% s2 )
                      + suppressWarnings( log( ( d1 %*% b + g %*% c )^2 + d1 %*% s2 ) ) )
    v  =  0.5 * colSums( ( y %*% d2 )^2 )

    sel  =  s2 > 0
    v[ sel ]  =  v0[ sel ]
    v
}

File: matlab/IA_fit.m
function v = IA_fit(abc,y,g)

a = abc(1,:);
b = abc(2,:);
c = abc(3,:);
s2 = (1-a.^2-b.^2-c.^2);
d1 = ones(length(y),1);
d2 = ones(1,length(a));

v0 = (1/2)*sum((y*d2-g*a).^2./((d1*b+g*c).^2 + d1*(s2))+log((d1*b+g*c).^2 + d1*(s2)),1); % minus log-likelihood
v = (1/2)*sum((y*d2).^2,1);
sel = find(s2>0);

% sel = find(min((d1*b+g*c).^2 + d1*(s2),[],1)>0);

v(sel) = v0(sel);

%%%%

